app/controllers/application_controller.rb

```diff
@@ -53,10 +53,8 @@ def set_announcements
   # an insufficient disk resource
   def set_my_quotas
     @my_quotas =
-      Quota.find(OodSupport::User.new).map do |quota|
-        SufficientQuotaDecorator.new(quota)
-      end.map do |quota|
-        QuotaPresenter.new(quota)
+      Quota.find(::Configuration.quota_paths, OodSupport::User.new).map do |quota|
+        QuotaPresenter.new(quota, threshold: ::Configuration.quota_threshold)
```

Comment:
See this comment: https://github.com/OSC/ood-dashboard/pull/367/files#r193222458
One of the reasons this code is complex and has such a presenter is because the view wants to actually display several quota messages for displaying disk size and file size.

So we have this duplication all over the place. A final refactor stab could actually take a look at simplifying Quota class further, by instead of having both files and disk and percent total messages on files and disk, just having something like:

```ruby
class Quota
  attr_accessor :type, :resource, :user_usage, :total_usage, :units
end
```

More objects would be created (2 for each json hash instead of 1) but you could probably get by without a presenter then. If you do an exploration with this, I'd save it for last and do it on a separate (yet another) branch.

Reply:
Actually for this idea, it might also work to have `Quota`, `FileQuota < Quota`, `BlockQuota < Quota`.